syntax = "proto3";
package EmailService;
option go_package = "streamflow/services/email/pb;pb";


service EmailService {
  rpc SendWelcomeEmail (WelcomeEmailRequest) returns (EmailResponse);
  rpc SendInvoiceEmail (SendInvoiceEmailRequest) returns (SendInvoiceEmailResponse);
  rpc SendPasswordUpdatedEmail (PasswordEmailRequest) returns (EmailResponse);
}

message WelcomeEmailRequest {
  string email = 1;
  string name = 2;
}

message SendInvoiceEmailRequest {
  string user_email = 1;
  int64 invoice_id = 2;
  string amount = 3;
  string status = 4;
}

message SendInvoiceEmailResponse {
  bool success = 1;
}

message PasswordEmailRequest {
  string user_email = 1;
  string user_name = 2;
}

message EmailResponse {
  bool success = 1;
  string message = 2;
}
